generator/internal/dart/dart.go

```diff
@@ -29,21 +29,13 @@ import (
 //go:embed templates
 var dartTemplates embed.FS
 
-// A map of message ID => name renames.
-//
-// `Duration` in particular is important to rename as this would conflict with
-// the `Duration` class in the 'dart:core' library (imported by default into
-// every library).
-// TODO(#1034): Renaming this class is practical but unlikely to be what we want
-// to do long term. We want a reliable, low-ceremony way of avoiding name
-// conflicts with dart:core types.
-var messageRenames = map[string]string{
-	".google.protobuf.Duration": "PbDuration",
-}
-
 var typedDataImport = "dart:typed_data"
 var httpImport = "package:http/http.dart"
 
+var needsCtorValidation = map[string]string{
+	".google.protobuf.Duration": ".google.protobuf.Duration",
+}
```

Comment:
This is probably fine, but maybe we should make that a configuration parameter in the codec? OTOH, I think only Duration, Timestamp, and (maybe) FieldMask will need it, so a hard-coded list may be perfectly fine.